src/main/java/org/embulk/input/kintone/KintoneInputPlugin.java

```diff
@@ -113,4 +142,95 @@ protected KintoneClient getKintoneClient()
     {
         return new KintoneClient();
     }
+
+    private List<Record> expandSubtable(final Record originalRecord, final List<String> subTableFieldCodes)
+    {
+        ArrayList<Record> records = new ArrayList<>();
+        records.add(cloneRecord(originalRecord));
+        for (String fieldCode : subTableFieldCodes) {
+            List<TableRow> tableRows = originalRecord.getSubtableFieldValue(fieldCode);
+            for (int idx = 0; idx < tableRows.size(); idx++) {
+                if (records.size() < idx + 1) {
+                    records.add(cloneRecord(originalRecord));
+                }
+
+                TableRow tableRow = tableRows.get(idx);
+                Record currentRecord = records.get(idx);
+                Set<String> tableFieldCodes = tableRow.getFieldCodes();
+                for (String tableFieldCode : tableFieldCodes) {
+                    currentRecord.putField(tableFieldCode, tableRow.getFieldValue(tableFieldCode));
+                }
+            }
+        }
+        return records;
+    }
+
+    private Record cloneRecord(final Record src)
+    {
+        Record dst = new Record(src.getId(), src.getRevision());
+        for (String fieldCode : src.getFieldCodes(true)) {
+            dst.putField(fieldCode, src.getFieldValue(fieldCode));
+        }
+        return dst;
+    }
+
+    private Schema buildSchema(final PluginTask task)
+    {
+        KintoneClient client = getKintoneClient();
+        client.validateAuth(task);
+        client.connect(task);
+
+        Map<String, FieldProperty> fields = client.getFields(task);
+        Builder builder = Schema.builder();
+
+        // built in schema
+        builder.add("$id", Types.LONG);
+        builder.add("$revision", Types.LONG);
+
+        for (Map.Entry<String, FieldProperty> fieldEntry : fields.entrySet()) {
```

Comment:
Since it is hash map and is not ordered, I think it would be better to sort by field code or something, for users easy to understand.

Reply:
Thank you.
I fixed it: [98bc585](https://github.com/trocco-io/embulk-input-kintone/pull/37/commits/98bc58531585bf920054214102b284d94120fb77)